File: src/main/scala/quickscala/chapter3/Example1.scala
package quickscala.chapter3

import utils.PlUtils

import scala.util.Random

/*
 * 3.1
 * 编写一段代码，将a设置为一个n个随机整数的数组，要求随机数介于0和n之间。
 */
object Example1 {

  def main(args: Array[String]): Unit = {

    val arr = randomNum(5)

    PlUtils.pri(arr.mkString("[",",","]"))

  }


  def randomNum(n:Int): Array[Int] ={
    require(n>0)
    (for(i<- 0 to n)yield {Random.nextInt(n)}).toArray
  }

}

File: src/main/java/quickjava/chapter3/Example1.java
package quickjava.chapter3;

import java.util.concurrent.ThreadLocalRandom;

/*
 * 3.1
 * 编写一段代码，将a设置为一个n个随机整数的数组，要求随机数介于0和n之间。
 */
public class Example1 {

    public static void main(String args[]){

        int[] arr = randomNum(5);
        for(int i :arr){
            System.out.println(i);
        }

    }

    /*
     * 3.1
     * 编写一段代码，将a设置为一个n个随机整数的数组，要求随机数介于0和n之间。
     */
    public static int[] randomNum(int n){
        if(n<0){
            return null;
        }
        int[] arr = new int[n];
        for(int i=0;i<n;i++){
            arr[i] = ThreadLocalRandom.current().nextInt(n);
        }
        return arr;
    }

}
